src/rules/no-extraneous-dependencies.js

```diff
@@ -162,7 +162,6 @@ function reportIfMissing(context, deps, depsOptions, node, name) {
   // Do not report when importing types
   if (
     node.importKind === 'type' ||
-    (node.parent && node.parent.importKind === 'type') ||
```

Comment:
Relatedly this was introduced [here](https://github.com/import-js/eslint-plugin-import/pull/1820/files#diff-4c4c694d76da5ddc56956bb432ce2bc90fbbea0404ee005e66ef5f4284bf5a94R114) and rendered obsolete [here](https://github.com/import-js/eslint-plugin-import/commit/93e060f63f3c6f557c7fea9cddc0018c033b058d#diff-4c4c694d76da5ddc56956bb432ce2bc90fbbea0404ee005e66ef5f4284bf5a94R210), I think.

[This comment](https://github.com/import-js/eslint-plugin-import/issues/1618#issuecomment-643082389) is describing `node.source`, which is what `node` used to be, prior to #1237. That requires the `node.parent.importKind` term. After, it's the ImportDeclaration/ImportExpression, which satisfies the `node.importKind` term instead.